lib/generators/manageiq/provider/provider_generator.rb

```diff
@@ -25,23 +29,51 @@ def create_provider_files
 
     alias provider_name file_name
 
+    def self.manager_types
+      @manager_types ||= {
+        "automation"    => "AutomationManager",
+        "cloud"         => "CloudManager",
+        "configuration" => "ConfigurationManager",
+        "container"     => "ContainerManager",
+        "infra"         => "InfraManager",
+        "monitoring"    => "MonitoringManager",
+        "network"       => "NetworkManager",
+        "physical"      => "PhysicalInfraManager",
+        "provisioning"  => "ProvisioningManager",
+        "storage"       => "StorageManager"
+      }
+    end
+
+    def validate_manager_type!
+      return unless manager_type.nil?
+      raise "Invalid manager_type: #{options[:manager_type]}\nMust be one of #{self.class.manager_types.keys.join(", ")}"
+    end
+
+    def manager_type
+      @manager_type ||= self.class.manager_types[options[:manager_type]]
+    end
+
+    def manager_path
+      manager_type.underscore
+    end
+
     def create_dummy
-      template "app/models/%plugin_path%/cloud_manager/event_catcher/runner.rb"
-      template "app/models/%plugin_path%/cloud_manager/event_catcher/stream.rb"
-      template "app/models/%plugin_path%/cloud_manager/metrics_collector_worker/runner.rb"
-      template "app/models/%plugin_path%/cloud_manager/refresh_worker/runner.rb"
-      template "app/models/%plugin_path%/cloud_manager/event_catcher.rb"
-      template "app/models/%plugin_path%/cloud_manager/metrics_capture.rb"
-      template "app/models/%plugin_path%/cloud_manager/metrics_collector_worker.rb"
-      template "app/models/%plugin_path%/cloud_manager/refresh_worker.rb"
-      template "app/models/%plugin_path%/cloud_manager/refresher.rb"
-      template "app/models/%plugin_path%/cloud_manager/vm.rb"
-      template "app/models/%plugin_path%/inventory/collector/cloud_manager.rb"
-      template "app/models/%plugin_path%/inventory/parser/cloud_manager.rb"
+      template "app/models/%plugin_path%/%manager_path%/event_catcher/runner.rb"
+      template "app/models/%plugin_path%/%manager_path%/event_catcher/stream.rb"
+      template "app/models/%plugin_path%/%manager_path%/metrics_collector_worker/runner.rb"
+      template "app/models/%plugin_path%/%manager_path%/refresh_worker/runner.rb"
+      template "app/models/%plugin_path%/%manager_path%/event_catcher.rb"
+      template "app/models/%plugin_path%/%manager_path%/metrics_capture.rb"
+      template "app/models/%plugin_path%/%manager_path%/metrics_collector_worker.rb"
+      template "app/models/%plugin_path%/%manager_path%/refresh_worker.rb"
+      template "app/models/%plugin_path%/%manager_path%/refresher.rb"
+      template "app/models/%plugin_path%/%manager_path%/vm.rb"
```

Comment:
It doesn't make sense for all manager types to have a Vm subclass, it'd be nice to only create this for e.g. cloud/infra manager types but the inventory refresh examples are based on creating VMs so we might want to tackle this later?

Reply:
yeah, I'm fine with tackling the direct classes later.  I think we'll need some sort of lookup table for manager type -> default objects.